library/std/Cargo.toml

```diff
@@ -21,6 +21,11 @@ compiler_builtins = { version = "0.1.35" }
 profiler_builtins = { path = "../profiler_builtins", optional = true }
 unwind = { path = "../unwind" }
 hashbrown = { version = "0.9.0", default-features = false, features = ['rustc-dep-of-std'] }
+# RDRAND feature is needed for SGX targets, but will work on other x86(-64) targets
+# unsupported in `getrandom` by default
```

Comment:
RDRAND breaks usage of rr.

Reply:
Are you complaining that a source of randomness in a process explicitly designed to be non-deterministic breaks reproducible executions? Or maybe I'm misunderstanding you completely here.

Reply:
Correct, rr depends on execution of instructions being deterministic. It can only record non-determinism originating from the OS through syscalls or interrupts. It can't record non-determinism caused by rdrand, causing rr to crash due to inconsistencies between the recorded and replayed trace.

Reply:
Currently the RDRAND backend will be only used on SGX, UEFI, and Hermit targets. Enabling the `rdrand` feature has absolutely no effect on other x86 targets. Note that `std` already [uses](https://github.com/rust-lang/rust/blob/master/library/std/src/sys/sgx/mod.rs#L146-L156) RDRAND for SGX targets, so changes in this PR should not cause any regressions outside of UEFI and Hermit targets, which currently simply use constant seeds.

Reply:
> Enabling the rdrand feature has absolutely no effect on other x86 targets (i.e. we always use OS API if possible).

Ah, ok.